File: app/src/main/java/com/harishsk/saveblood/ListViewAdapter.java
package com.harishsk.saveblood;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.List;

public class ListViewAdapter extends BaseAdapter {

    private Activity activity;
    private List<Blood> BlList;

    ListViewAdapter(Activity activity, List<Blood> blList) {
        this.activity = activity;
        BlList = blList;
    }

    @Override
    public int getCount() {
        return BlList.size();
    }

    @Override
    public Object getItem(int position) {
        return BlList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) activity.getBaseContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View itemview = inflater.inflate(R.layout.list_adapter_data, null);

        TextView name = itemview.findViewById(R.id.lt_name);
        TextView gender = itemview.findViewById(R.id.lt_gender);
        TextView age = itemview.findViewById(R.id.lt_age);
        TextView blgrp = itemview.findViewById(R.id.lt_blgrp);
        TextView phno = itemview.findViewById(R.id.lt_phno);
        TextView place = itemview.findViewById(R.id.lt_place);

        name.setText(BlList.get(position).getName());
        gender.setText(BlList.get(position).getGender());
        age.setText(BlList.get(position).getAge());
        blgrp.setText(BlList.get(position).getBlgrp());
        phno.setText(BlList.get(position).getPhno());
        place.setText(BlList.get(position).getPlace());

        return itemview;
    }
}

File: app/src/main/java/com/harishsk/saveblood/adapter/ListViewAdapter.kt
package com.harishsk.saveblood.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.harishsk.saveblood.R
import com.harishsk.saveblood.model.Blood

class ListViewAdapter internal constructor(
    private val activity: AppCompatActivity?,
    private val BlList: List<Blood>
) : BaseAdapter() {
    override fun getCount(): Int {
        return BlList.size
    }

    override fun getItem(position: Int): Any {
        return BlList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(
        position: Int,
        convertView: View,
        parent: ViewGroup
    ): View {
        val inflater = activity!!.baseContext
            .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val itemview = inflater.inflate(R.layout.list_adapter_data, null)
        val name = itemview.findViewById<TextView>(R.id.lt_name)
        val gender = itemview.findViewById<TextView>(R.id.lt_gender)
        val age = itemview.findViewById<TextView>(R.id.lt_age)
        val blgrp = itemview.findViewById<TextView>(R.id.lt_blgrp)
        val phno = itemview.findViewById<TextView>(R.id.lt_phno)
        val place = itemview.findViewById<TextView>(R.id.lt_place)
        name.text = BlList[position].name
        gender.text = BlList[position].gender
        age.text = BlList[position].age
        blgrp.text = BlList[position].blgrp
        phno.text = BlList[position].phno
        place.text = BlList[position].place
        return itemview
    }

}